protocol/x/listing/keeper/listing.go

```diff
@@ -148,3 +148,24 @@ func (k Keeper) CreatePerpetual(
 
 	return perpetual.GetId(), nil
 }
+
+// Function to set listing vault deposit params in module store
+func (k Keeper) SetListingVaultDepositParams(
+	ctx sdk.Context,
+	params types.ListingVaultDepositParams,
```

Comment:
can we add validation on this? e.g. ensure that `deposit_amount` is non-negative

Reply:
Shouldn't really matter since this is gov controlled. 
Also the transfer to vault functions in x/vault should already have this validation, I believe, which is the more important place

Reply:
usually it's nice to have validation even on gov controlled params and don't think we should depend on validation in another module